DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/KeywordCampaign.java

```diff
@@ -14,21 +14,22 @@
 
 package com.google.sps.servlets;
 
-import java.io.IOException;
-import java.io.PrintWriter;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
-@WebServlet("/")
-public class EmptyServlet extends HttpServlet {
+public class KeywordCampaign {
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    }
+    public int keywordCampaignId;
+    public int userId;
+    public int impressions;
+    public int clicks;
+    public int cost;
```

Comment:
Two asks:
1. Can you describe the main classes in the design doc (keyword campaign, dsa campaign, uplift estimation stats, etc.) so it is clear what information will be passed around?
2. What is the purpose of the KeywordCampaign class? From my understanding of the doc, it is to take campaign settings from and to use as defaults when creating the dsa campaign as it is essentially the advertisers original input. But here there aren't any such fields. And what do you plan on doing with the impressions/clicks/costs here?

Reply:
1. The other day I created a data section in the design doc explaining the information in the classes.
2. Yes, that is the purpose of the keyword campaign class. Right now, we don't have any input fields as this is just a bare-bones class to make sure we can get everything connected, but we will most likely add input fields in the future. We have impressions/clicks/costs because at the end of the estimation, we are displaying uplift which is (DSA simulation results) - (keyword campaign results), so to find the change we will need initial impression/clicks/costs values.

Reply:
Ok that makes sense. In that case can you add a TODO comment explaining that these setting fields need to be added.

Also, the exact way that the stats are computed are up to you. But for reference in our system, the raw uplift stats is computed in the backend, we generally use the campaign's stats to see how much percent uplift gain we're getting.